services/navigation/builtin/builtin.go

```diff
@@ -348,16 +389,28 @@ func (svc *builtIn) nextWaypoint(ctx context.Context) (navigation.Waypoint, erro
 }
 
 func (svc *builtIn) waypointReached(ctx context.Context) error {
-	wp, err := svc.nextWaypoint(ctx)
-	if err != nil {
-		return fmt.Errorf("can't mark waypoint reached: %w", err)
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
```

Comment:
I don't know that this is useful here. This isn't a long running function, is it?

Reply:
No, but the thinking is that if the context has been canceled, we don't want the waypoint to be updated as reached.

Reply:
Fair enough. Probably not quite how I'd write it, but no harm as-is then.